### PR TITLE
Fortran test for SLE Toolchain Module's GCC5

### PR DESCRIPTION
**Pull request by mnowaksuse**

### CODE REVIEW

File: tests/toolchain/gcc5_fortran_compilation.pm [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# without any warranty.
+
+use base "opensusebasetest";
+use strict;
+use testapi;
+
+sub run() {
+    my $self = shift;
+
+    select_console('root-console');
```
Comment (dzedro) [inline tests/toolchain/gcc5_fortran_compilation.pm]: this is not needed if this test will be part of existing toolchain tests

File: tests/toolchain/gcc5_fortran_compilation.pm [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    script_run "patch -p0 < ../adapt-FM406-to-fortran-95.patch";
+
+    # Build
+    assert_script_run './driver_parse', 120;
+    # Test
+    assert_script_run './driver_run', 300;
+    
+    assert_script_run 'if ! grep " [1-9][0-9]* TESTS FAILED" *.res; then true; else false; fi';
+    
+    sleep 60;
```
Comment (okurz) [inline tests/toolchain/gcc5_fortran_compilation.pm]: you shouldn't use explicit sleeps. Why do you add this? Can you wait for something else to happen? e.g. if the script run was successful or do an assert_screen with a long timeout?

Comment (dzedro) [inline tests/toolchain/gcc5_fortran_compilation.pm]: if Michal tested it and he knows that default timeout is not enough, then I would say any longer timeout is OK, I used 4000 for C++ compilation.

File: tests/toolchain/gcc5_fortran_compilation.pm (hunk 1 above)
Comment (mnowaksuse) [inline tests/toolchain/gcc5_fortran_compilation.pm]: You mean just "select_console"? Right, that's a stray from my manual testing. Will remove it.

File: tests/toolchain/gcc5_fortran_compilation.pm (hunk 2 above)
Comment (mnowaksuse) [inline tests/toolchain/gcc5_fortran_compilation.pm]: Right, it's another stray from my manual testing (just wanted to see the output on VNC for a longer time); shouldn't be there. Will remove it.

Comment (dzedro) [inline tests/toolchain/gcc5_fortran_compilation.pm]: yes, better change assert_script_run timeout

File: tests/toolchain/gcc5_fortran_compilation.pm [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    script_run 'echo "exit \${failed}" >> driver_parse';
+
+    script_run "patch -p0 < ../adapt-FM406-to-fortran-95.patch";
+
+    # Build
+    assert_script_run './driver_parse', 120;
+    # Test
+    assert_script_run './driver_run', 300;
+    
+    assert_script_run 'if ! grep " [1-9][0-9]* TESTS FAILED" *.res; then true; else false; fi';
```
Comment (okurz) [inline tests/toolchain/gcc5_fortran_compilation.pm]: Wouldn't it be enough to just call
```
assert_script_run 'grep " [1-9][0-9]* TESTS FAILED" *.res';
```
The grep will just return the correct exit code, isn't it?

File: tests/toolchain/gcc5_fortran_compilation.pm (hunk 1 above)
Comment (dzedro) [inline tests/toolchain/gcc5_fortran_compilation.pm]: yes

File: tests/toolchain/gcc5_fortran_compilation.pm (hunk 3 above)
Comment (mnowaksuse) [inline tests/toolchain/gcc5_fortran_compilation.pm]: I need the `assert_script_run` to fail when grep finds something, and pass when not found. grep returns by default exit code (0 ... when found, 1 ... when not found) the opposite way I need. Unless there's an option for grep which forces grep to return 0 when the pattern is not found..

Comment (aaannz) [inline tests/toolchain/gcc5_fortran_compilation.pm]: How about `assert_script_run '! grep " [1-9][0-9]* TESTS FAILED" *.res';`

Comment (mnowaksuse) [inline tests/toolchain/gcc5_fortran_compilation.pm]: Right, makes sense :).

**Comment by mnowaksuse** [comment]
All the requested changes should be here. Should I pay attention to Travis CI failed check?

**Comment by aaannz** [comment]
Call `tools/tidy` to tidy your code and to pass travis check.

**Comment by mnowaksuse** [comment]
Travis CI tamed.

**Comment by okurz** [comment]
@mnowaksuse everything travis CI does can also be done by you locally. Call/see `make check` for fast feedback before even submitting and waiting for travis.

File: tests/toolchain/gcc5_fortran_compilation.pm [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+sub run() {
+    my $self = shift;
+
+    my $package_main  = data_url('toolchain/fcvs21_f95.tar.bz2');
+    my $package_fix   = data_url('toolchain/FM923.DAT');
+    my $package_patch = data_url('toolchain/adapt-FM406-to-fortran-95.patch');
+    script_run "wget $package_main";
+    script_run "wget $package_fix";
+    script_run "wget $package_patch";
```
Comment (coolo) [inline tests/toolchain/gcc5_fortran_compilation.pm]: I think you want assert_script_run - or your tests will depend on magic sleeps.

**Comment by sysrich** [comment]
Can you rebase/squash these commits? 

http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

eg. you probably want to do something like 'git rebase -i HEAD~6' and then replace 'pick' with 's' for 'squash' for the 5 commits after your original one (and probably also removing the commit messages from all apart from the first commit also)

This way when we merge the pull request we only have the commit and comments for the issue you were intending to fix, not the tuneup :)

**Comment by mnowaksuse** [comment]
Thanks Rich, all the bits should be there and in the right format.

**Pull request merged**